colcon_output/event_handler/event_log.py

```diff
@@ -39,21 +40,23 @@ def __call__(self, event):  # noqa: D102
             members = data.__dict__
         except AttributeError:
             members = {s: getattr(data, s) for s in data.__slots__}
-        with self._path.open(mode='a') as h:
-            h.write(
-                '[%f] (%s) %s: %s\n' % (
-                    self._get_relative_time(), context,
-                    data.__class__.__name__, members))
+        self._file_handle.write(
+            '[%f] (%s) %s: %s\n' % (
+                self._get_relative_time(), context,
+                data.__class__.__name__, members))
+        self._file_handle.flush()
+
+        if isinstance(data, EventReactorShutdown):
+            self._file_handle.close()
```

Comment:
Should we worry about a scenario where we never receive a shutdown event and the file is not closed?

Reply:
While that could happen in case Python exits early e.g. with an unhandled exception the file will be close automatically when the Python process ends.

Since each line is already flushed I don't see a problem with that.